lib/Doctrine/DBAL/Platforms/SQLServer2012Platform.php

```diff
@@ -143,4 +131,35 @@ protected function doModifyLimitQuery($query, $limit, $offset = null)
 
         return $query;
     }
+
+    /**
+     * @param string $query
+     */
+    protected function shouldAddOrderBy($query): bool
+    {
+        // Find the position of the last instance of ORDER BY and ensure it is not within a parenthetical statement
+        // but can be in a newline
+        $matches      = [];
+        $matchesCount = preg_match_all('/[\\s]+order\\s+by\\s/im', $query, $matches, PREG_OFFSET_CAPTURE);
+        if ($matchesCount === 0) {
+            return true;
+        }
+
+        // ORDER BY instance may be in a subquery after ORDER BY
+        // e.g. SELECT col1 FROM test ORDER BY (SELECT col2 from test ORDER BY col2)
+        // if in the searched query ORDER BY clause was found where
+        //  number of open parentheses after the occurrence of the clause is equal to
+        //  number of closed brackets after the occurrence of the clause,
```

Comment:
Is extra indentation here intended or it's a typo?

Reply:
I did this on purpose to highlight the condition. Is it wrong?

Reply:
It doesn't look highlighting to me. If you want it highlighted, you could probably use some punctuation or markup. Otherwise, just leave it as plain tex.